server/utils/ethereum.js

```diff
@@ -0,0 +1,44 @@
+const Web3 = require('web3')
+const ipfs = require('./ipfs')
+const storageJSON = require('../../build/contracts/Storage.json')
+const accounts = require('../../accounts.json')
+
+const web3 = new Web3(
+  new Web3.providers.WebsocketProvider('ws://localhost:8545')
+)
+
+const registerWatcher = contract => {
+  contract.events.Registered({}, async (err, event) => {
+    if (err) console.error('Error subscribing: ', err)
+
+    ipfs
+      .getAndPin(event.returnValues.cid)
+      .then(res => {
+        if (!res[0]) console.error(res)
+        else console.log('Pinned', res[0].hash)
+      })
+      .catch(error => {
+        console.error('Error pinning: ', error)
+      })
+  })
+}
+
+// TODO: Move to a test file
```

Comment:
should we just do this before merging?

Reply:
We'd need to figure out how to get Jon's script working for tests first.